Improve ENSRainbow Docs (#254)
namehash/ensnode: apps/ensrainbow/README.md

ENSRainbow is an ENSNode sidecar service for healing ENS labels. It provides a simple API endpoint to heal ENS labelhashes back to their original labels.

## Prerequisites

- Docker installed on your system
- Node.js v18 or later (for local development) — see monorepo package.json for engines spec

## System Requirements

### Data Ingestion Requirements (`ingest` command)
- **Storage**:
  - At least 15 GB of free disk space:
    - 6.37 GB for the compressed rainbow tables download
    - ~7 GB for the LevelDB database after ingestion
    - Additional temporary space during build/ingestion
- **Memory**: At least 4 GB RAM recommended

> ### Server Requirements (`serve` command)

Reviewer: ```suggestion
### API Server Requirements (`serve` command)
```

- Is maintained and monitored by the NameHash Labs team
- Runs the latest version of ENSRainbow

> **Important**: While we provide a freely available hosted instance, we strongly recommend running your own ENSRainbow instance alongside ENSNode in your infrastructure. Co-locating these services on the same local network significantly improves indexing performance, reduces latency, ensures guaranteed uptime, and maintains data privacy. This setup is particularly important for production environments.

## API Endpoints

### Health Check

```bash
curl https://api.ensrainbow.io/health
```

Response: `{"status":"ok"}`

### Heal Label

```bash
curl https://api.ensrainbow.io/v1/heal/0x[labelhash]
```

> The `labelhash` parameter must be strictly formatted according to these requirements:

Reviewer: ```suggestion
The `labelhash` parameter must be strictly formatted as a "normalized labelhash" according to these requirements:
```

- Must start with '0x'
- Must be exactly 66 characters long (including '0x' prefix)
- Must be lowercase
- Must be a valid hex string that converts to exactly 32 bytes

> For example, this is a valid labelhash:

Reviewer: ```suggestion
For example, this is a valid "normalized labelhash":
```

## Environment Variables

The following environment variables can be used to configure different aspects of the service:

### Global Variables
These variables affect all commands:
- `DATA_DIR`: Default directory for the LevelDB database. If not provided, defaults to `data/`
- `LOG_LEVEL`: Logging level for all operations, one of: "debug", "info", "warn", "error". If not provided, defaults to `info`

### Server Command Variables
These variables affect the ENSRainbow server operation:

> - `PORT`: Server port (default: 3223)

Author: I think `PORT` should be available by env var so you can change while running docker image without changing the code.

> - `PORT`: Server port (default: 3223)

Reviewer: @djstrong Ah I see! I also see the work you did on `getDefaultPort `. Cool! As I understand, if a port is passed on the command line, it is given priority over a port set through the `PORT` env var?

Is that right? Suggest documenting how the port actually used is decided.

Or maybe raise an error if both the env var is set and the commandline arg are passed and the values don't agree? I'm a fan of failing loud when people do bad things. If we enforce that these values can't both be defined with conflicting values then I think no need to add more special documentation about this, since if someone tries to do something silly, they just get an error 😄

### Global Variables
These variables affect all commands:

> - `LOG_LEVEL`: Logging level for all operations, one of: "debug", "info", "warn", "error". If not provided, defaults to `info`

Reviewer: Sorry if I'm going across different PRs. Perhaps we should add the extra log level here for silent? Please feel welcome to ignore this feedback if not a good idea based on separate PRs / merge strategy / etc 😅

> - `NODE_ENV`: Standard environment variable (used for logging format)

Reviewer: ```suggestion
- `NODE_ENV`: Standard Node.js environment variable used to indicate the current environment, such as "development, "test", or "production". If "production" a performance optimized logging format is used.
```
